portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -32,9 +33,34 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     messages.add("first message");
     messages.add("second message");
     messages.add("third message");
-    Gson gson = new Gson();
-    String jsonMessage = gson.toJson(messages);
+    String jsonMessage = convertToJson(messages);
+    response.setContentType("application/json;");
+    response.getWriter().println(jsonMessage);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String email = getParameter(request, "email", null);
+    String name = getParameter(request, "name", null);
+    String message = getParameter(request, "message", null);
+    Message messageObj = new Message(name, email, message);
+    String jsonMessage = convertToJson(messageObj);
     response.setContentType("application/json;");
     response.getWriter().println(jsonMessage);
   }
+
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```

Comment:
Seems like the custom getParameter function doesn't do much over the original, since you always supply with default value as null.